tstest/natlab/firewall.go

```diff
@@ -5,26 +5,67 @@
 package natlab
 
 import (
+	"fmt"
 	"sync"
 	"time"
 
 	"inet.af/netaddr"
 )
 
-type session struct {
+// EndpointDependence expresses
+type FirewallSelectivity int
+
+const (
+	// APDF specifies a destination address-and-port dependent
+	// firewall. Outbound traffic to an ip:port authorizes traffic
+	// from that ip:port exactly, and nothing else.
+	APDF FirewallSelectivity = iota
+	// ADF specifies a destination address dependent firewall. Once
+	// outbound traffic has been seen to an IP address, that IP
+	// address can talk back from any port.
+	ADF
+	// EIF specifies a destination endpoint independent firewall. Once
+	// outbound traffic has been seen from a source, anyone can talk
+	// back to that source.
+	EIF
+)
+
+type fwKey struct {
```

Comment:
docs here too. mention that not all 4 fields (2x2) will be populated dependent on the selectivity